educe/stac/rfc.py

```diff
@@ -138,21 +139,22 @@ def violations(self):
         :rtype: [string]
         '''
         graph = self._graph
-        nodes = graph.first_outermost_dus()
-        res = list()
+        nodes = self._nodes
         if len(nodes) < 2:
-            return res
+            return list()
 
-        points = self._frontier_points(nodes)
-        nexts = itr.islice(nodes, 1, None)
-        for last, node1 in zip(nodes, nexts):
-            for lnk in graph.links(node1):
-                if not self._is_incoming_to(node1, lnk):
+        violations = list()
+        for i, new_node in enumerate(nodes):
+            last_node = nodes[i-1] if i>0 else None
```

Comment:
@kowey could I delegate the review of this portion of the code to you? I can't reconcile the intended semantics, the current version and the proposed new version.